### PR TITLE
Add facebook time function

### PR DESCRIPTION
**Pull request by super-ienien**
Hi,

this add a function to get the current facebook time. It's based on the date returned by facebook in the http headers of a request response. The delta is calculated against the current time machine.

### CODE REVIEW

**Comment by dantman** [comment]
Can you explain the purpose of this function returning the time from Facebook's perspective?

I should also note that this is not going to be reliable.
* The Date: header is set on Facebook's side at some time during the request, the response is then sent to you, and then you calculate the delta. You need to compensate for the time it takes for the response to travel from Facebook to your server or else your delta will be off.
* The Date: header does not include miliseconds, but `Date.now()` does, this is going to throw your delta off further.
* The delta is calculated every request, no matter what the request type, with no normalization. This means that if you make an expensive request that takes some time to get back to you (photo download, a POST upload if Date: is calculated before the end, or just a temporary network problem) it is going to introduce a long and incorrect delta until your next request.

Basically, if you have a server properly setup to use a ntp server which should match it up with the same universal time Facebook should be using, instead of this `FB.now()` returning the time Facebook uses it's just going to introduce an incorrect and unreliable delta that will ruin your already correct time calculations.

Code issues:
* Bad indentation and formatting
* Merge conflicts in the doc

**Comment by super-ienien** [comment]
Hi,

the purpose of this function is to use the facebook server time for making requests that the since parameter. For exemple if I wan't to get all comments of a facebook post that was published in the last couple of minutes, i will do : `.since(FB.now()/1000-120)`. I needed this kind of feature to be sure that my request is based on the facebook time, just in case that my server has not exactly the same time as facebook. I can't rely on my server time for what I wan't to do, even if the server is synced with ntp (ntp can fail too). That's not bad for me if the delta is not accurate to one second or two

If you think this is not a good feature. I don't care, I need it so I will keep working with my fork and rebase when you'll add new cool features.
If you think it could be merged but with some modifications and warnings in the documentation about the accuracy, I will do those modifications. I could for exemple, calculate the delta just once at the first request.

**Comment by dantman** [comment]
Your example is rather strange and doesn't sound like a normal use case of the graph api ("comments within last X minutes"; rather than what most people would do "X most recent comments" which doesn't require timing functions). I doubt that using the `Date:` header for time deltas is an intended way of using the graph api. If I were Facebook, I'd sooner support something like `since={now}-120s` than suggest using the `Date:` header.

Also, if you're happy to calculate the delta once at the start (or even periodically) instead of as part of every api call; you don't even need an official `FB.now()` function.

For that reason, I don't recommend bothering to maintain a fork.

Just make an `OPTIONS` http request (using a library such as `request`, which you're already indirectly depending on) to `https://graph.facebook.com/` and parse the `Date:` header. You don't need to do that as part of an actual API call. And doing so outside of API calls ensures you're not getting deltas corrupted by delays in generating, transmitting, or parsing the response body.
